### PR TITLE
Adding new sample notebook

### PR DESCRIPTION
**Pull request by rhea-j**
Adding a sample notebook that showcases how we can perform analysis on ArcGIS online. The notebook is about determining palm oil suitability. 

cc - @AtmaMani 
-----

# Checklist

Please go through each entry in the below checklist and mark an 'X' if that condition has been met. Every entry should be marked with an 'X' to be get the Pull Request approved.

- [x] All `import`s are in the first cell? First block of imports are standard libraries, second block are 3rd party libraries, third block are all `arcgis` imports? Note that in some cases, for samples, it is a good idea to keep the imports next to where they are used, particularly for uncommonly used features that we want to highlight.
- [x] All `GIS` object instantiations are one of the following?
    - `gis = GIS()`
    - `gis = GIS('home')`
    - `gis = GIS(profile="your_online_portal")`
    - `gis = GIS('https://pythonapi.playground.esri.com/portal')`
    - `gis = GIS(profile="your_enterprise_portal")`
- [x] If this notebook requires setup or teardown, did you add the appropriate code to `./misc/setup.py` and/or `./misc/teardown.py`?
- [x] If this notebook references any portal items that need to be staged on AGOL/Python API playground, did you coordinate with a Python API team member to stage the item the correct way with the api\_data\_owner user?
- [x] Code refactored & split out across multiple cells, useful comments?
- [x] Consistent voice/tense/narrative style? Thoroughly checked for typos?
- [x] All images used like `` instead of ``? All map widgets contain a static image preview? (Call `mapview_inst.take_screenshot()` to do so)
- [x] All file paths are constructed in an OS-agnostic fashion with `os.path.join()`? (Instead of `r"\foo\bar"`, `os.path.join(os.path.sep, "foo", "bar")`, etc.)
- [x] **IF YOU WANT THIS SAMPLE TO BE DISPLAYED ON THE DEVELOPERS.ARCGIS.COM WEBSITE**, ping @ DavidJVitale so he can add it to the list for the next deploy

### CODE REVIEW

**Comment by rhea-j** [comment]
@priyankatuteja This is the notebook that we had discussed about. The notebook uses ArcGIS online and is not meant to work in try it live mode.

**Comment by rhea-j** [comment]
@BP-Ent Could you go through this notebook?

File: samples/04_gis_analysts_data_scientists/determining_site_suitability_for_oil_palm_plantation.ipynb [hunk 1]
```diff
@@ -0,0 +1,640 @@
+{
```
Comment (BP-Ent) [inline samples/04_gis_analysts_data_scientists/determining_site_suitability_for_oil_palm_plantation.ipynb:1]: <ul><li>To run raster analysis functions on ArcGIS Online, you must have privileges to create and publish content and to perform Raster Analysis.</li></ul><p> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/1039/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='5'/>

Comment (BP-Ent) [inline samples/04_gis_analysts_data_scientists/determining_site_suitability_for_oil_palm_plantation.ipynb:1]: <p>In order to carry out the suitability analysis, we need to consider some of the favorable environmental and crop criteria settings: </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/1039/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='11'/>

Comment (BP-Ent) [inline samples/04_gis_analysts_data_scientists/determining_site_suitability_for_oil_palm_plantation.ipynb:1]: <p>We can visualize the function chain and help document specific to this RFT in order to understand the input parameters and the functions involved.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/1039/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='16'/>

Comment (BP-Ent) [inline samples/04_gis_analysts_data_scientists/determining_site_suitability_for_oil_palm_plantation.ipynb:1]: <p>A '?' before or after an RFT object will display the help document consisting of the parameters marked as public by the author of the RFT.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/1039/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='18'/>

Comment (BP-Ent) [inline samples/04_gis_analysts_data_scientists/determining_site_suitability_for_oil_palm_plantation.ipynb:1]: <p>In the output, the red areas denote places that are less suitable for oil palm plantation, while green areas represent places that are more suitable. </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/1039/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='37'/>

**Review by BP-Ent — CHANGES_REQUESTED** [review]
Suggested changes made on ReviewNB

**Comment by rhea-j** [comment]
@BP-Ent Thank you for going through it. I have made the changes as per your suggestions.

File: samples/04_gis_analysts_data_scientists/determining_site_suitability_for_oil_palm_plantation.ipynb [hunk 2]
```diff
@@ -0,0 +1,639 @@
...
```
Comment (AtmaMani) [inline samples/04_gis_analysts_data_scientists/determining_site_suitability_for_oil_palm_plantation.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #1.    upload = arcgis.gis.ContentManager(gis)
</pre><p>Can we simplify the first two lines into this:</p><p><code>gis.content.add(item_properties={....})</code>?</p><p> </p><p>For notebooks that are samples, we typically provide a reproducible workflow. Hence, can you zip up and upload the rasters, RFT as an item on Geosaurus AGO org and refer to it in the notebook? Perhaps you can add a cell before this which downloads this item, unpacks (using zipfile lib) to a relative path which is fed to each of the copy_raster and this add_item call?</p><p> </p><p>You can publish the item using <code>api_data_owner</code>  account, delete protect it and share it to <code>everyone</code>  and refer to it using its item id</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/1039/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='13'/>

Comment (AtmaMani) [inline samples/04_gis_analysts_data_scientists/determining_site_suitability_for_oil_palm_plantation.ipynb:1]: <p>The <code>input_raster</code>  parameter is not clear to me. How are you able to point to the same folder but copy different files each time? Is the <code>output_name</code>  used to determine what input file is picked to be published?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/1039/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='24'/>

**Review by AtmaMani — CHANGES_REQUESTED** [review]
@rhea-j this is a nice sample, shows the RFT functionality very well. I have requested for some changes

**Comment by rhea-j** [comment]
@AtmaMani Thank you for reviewing. have made the necessary changes. Could you review it again?

**Review by AtmaMani — APPROVED** [review]
Revisions look great! thanks

**Pull request merged**